### DISCUSSION TITLE
How can I retirieve the domain name from a connection object?

### DISCUSSION THREAD

**Discussion by voger** [xmppjs/xmpp.js#887] [Q&A]
As in connection example 

```javascript
const { client, xml } = require("@xmpp/client");
const debug = require("@xmpp/debug");

const xmpp = client({
  service: "ws://localhost:5280/xmpp-websocket",
  domain: "localhost",  // <---- This one?
  resource: "example",
  username: "username",
  password: "password",
});
```

How do I later retrieve that domain name in the code?

**Comment by sonnyp** [comment]
Out of curiosity, why do you need to retrieve the domain if you provide it?

Ask differently, why should xmpp.js expose an API to give you back what you provided ?

**Reply by voger** [reply]
I am writing an XMPP console web app for fun and learning and I have this idea to add a tree with disco items and info. I want to name the root of the tree after the domain. The address is provided in another part of the app so I want somehow to retrieve it in this part of the app also. I could pass it from where I collected it in the first place but I would prefer to have it from the connection itself. 

btw: after I posted this question I discovered this 

```javascript
connection.entity.options.domain
```

but I don't know if there is a better way to retrieve it.

**Reply by sonnyp** [reply]
`connection.entity.options.domain` is not documented and might stop working 

xmpp.js should expose the domain though; can you open an issue?

I'd suggest storing the domain yourself until then.

**Reply by voger** [reply]
Thank you. I opened an issue #888
